CHANGELOG.md

```diff
@@ -4,6 +4,28 @@ The format is based on [Keep a Changelog].
 
 [Keep a Changelog]: http://keepachangelog.com/en/1.0.0/
 
+## 0.5.0
+
+This release shifts `scale-decode` to being a mirror of a new `scale-encode` crate, and:
+- Adds a new `IntoVisitor` trait that types can implement if there is a `Visitor` which can be used to decode
+  into them.
+- Adds a new `DecodeAsType` trait to mirror the `EncodeAsType` trait there; any type that implements `IntoVisitor`
+  automatically implements `DecodeAsType`.
+- Adds a new `DecodeAsFields` trait to mirror `EncodeAsFields`, implemented for tuple and struct types.
+- Moves the `Visitor` trait into a sub module and re-works the interface to allow zero copy decoding, allow more
+  concise implementations of it, and provide a fallback escape hatch to allow for more arbitrary `DecodeAsType`
+  implementations from it.
+- Implements `DecodeAsType` (via `Visitor` and `IntoVisitor` impls) and `DecodeAsFields` on common types.
+- Adds a `DecodeAsType` derive macro to auto-generate impls on custom struct and enum types.
+
+Any `Visitor` impls wil need to be updated to use the refined `Visitor` trait; this should be fairly mechanical
```

Comment:
```suggestion
Any `Visitor` impls will need to be updated to use the refined `Visitor` trait; this should be fairly mechanical
```

Reply:
Good spot; I re read this twice and didn't notice that!